File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
 int countnodes(struct TreeNode* root){
    if(root==NULL){
        return 0;
    }
    return 1+ countnodes(root->left)+countnodes(root->right);
 }
 void inorder(struct TreeNode*root,int*arr,int *i){
    
    if(root==NULL){
        return ;
    }
    inorder(root->left,arr,i);
    
    arr[*i]=root->val;
    (*i)++;
    inorder(root->right,arr,i);

    


 }
 
int kthSmallest(struct TreeNode* root, int k) {
    int size=countnodes(root);
    int *arr=malloc(sizeof(int)*size);
    int i=0;
    inorder(root,arr,&i);
    return arr[k-1];
    
}

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        # def countnodes(root):
        #     if root is None:
        #         return 0
        #     else:
        #         return 1+countnodes(root.left)+countnodes(root.right)
        def inorder(root, arr):
            if root is None:
                return 0
            inorder(root.left,arr)
            arr.append(root.val)
            inorder(root.right,arr)
        arr=[]
        inorder(root,arr)
        return arr[k-1]

        